ush/setup.py

```diff
@@ -1082,78 +1062,22 @@ def get_location(xcs, fmt, expt_cfg):
-        if dt_subhourly_post_mnts == 0:
-            logger.warning(
-                f"""
-                When performing sub-hourly post (i.e. SUB_HOURLY_POST set to \"TRUE\"),
-                DT_SUBHOURLY_POST_MNTS must be set to a value greater than 0; otherwise,
-                sub-hourly output is not really being performed:
-                  DT_SUBHOURLY_POST_MNTS = \"{DT_SUBHOURLY_POST_MNTS}\"
-                Resetting SUB_HOURLY_POST to \"FALSE\".  If you do not want this, you
-                must set DT_SUBHOURLY_POST_MNTS to something other than zero."""
-            )
-            post_config["SUB_HOURLY_POST"] = False
-
-        if dt_subhourly_post_mnts < 1 or dt_subhourly_post_mnts > 59:
-            raise ValueError(
-                f'''
-                When SUB_HOURLY_POST is set to \"TRUE\",
-                DT_SUBHOURLY_POST_MNTS must be set to an integer between 1 and 59,
-                inclusive but:
-                  DT_SUBHOURLY_POST_MNTS = \"{dt_subhourly_post_mnts}\"'''
-            )
-
-        # Check that DT_SUBHOURLY_POST_MNTS (after converting to seconds) is
-        # evenly divisible by the forecast model's main time step DT_ATMOS.
-        dt_atmos = fcst_config["DT_ATMOS"]
-        rem = dt_subhourly_post_mnts * 60 % dt_atmos
-        if rem != 0:
-            raise ValueError(
-                f"""
-                When SUB_HOURLY_POST is set to \"TRUE\") the post
-                processing interval in seconds must be evenly divisible
-                by the time step DT_ATMOS used in the forecast model,
-                i.e. the remainder must be zero.  In this case, it is
-                not:
-
-                  DT_SUBHOURLY_POST_MNTS = \"{dt_subhourly_post_mnts}\"
-                  DT_ATMOS = \"{dt_atmos}\"
-                  remainder = (DT_SUBHOURLY_POST_MNTS*60) %% DT_ATMOS = {rem}
-
-                Please reset DT_SUBHOURLY_POST_MNTS and/or DT_ATMOS so
-                that this remainder is zero."""
-            )
+    post_config = expt_config["task_run_post"]
 
     # Make sure the post output domain is set
-    predef_grid_name = workflow_config.get("PREDEF_GRID_NAME")
-    post_output_domain_name = post_config.get("POST_OUTPUT_DOMAIN_NAME")
+    post_output_domain_name = post_config["post_output_domain_name"]
 
-    if not post_output_domain_name:
-        if not predef_grid_name:
-            raise Exception(
-                f"""
-                The domain name used in naming the run_post output files
-                (POST_OUTPUT_DOMAIN_NAME) has not been set:
-                POST_OUTPUT_DOMAIN_NAME = \"{post_output_domain_name}\"
-                If this experiment is not using a predefined grid (i.e. if
-                PREDEF_GRID_NAME is set to a null string), POST_OUTPUT_DOMAIN_NAME
-                must be set in the configuration file (\"{user_config}\"). """
-            )
-        post_output_domain_name = predef_grid_name
+    if "{{ " in post_output_domain_name:
```

Comment:
I get the idea, but is the a stronger way to define "valid post output domain name" than "doesn't contain a Jinja2 expression start token"? Is there e.g. a regex that would match a valid value that we could use here?

Reply:
This can literally be any string and we want to at least make sure that the jinja template got filled in.

Reply:
Well, apparently it can't be a string with `{{` in it...

Reply:
I'm gonna need help here. I don't love regex's and I find this very readable to indicate that we want to check for unrendered Jinja. I am legit having a hard time understanding how this line could fail us.

Support for my regex distaste in community applications:
* Regex's are hard to maintain bc they're hard to understand (a lot of people with different programming experience contribute to this repo)
* The syntax is different between each language (https://www.greenend.org.uk/rjk/tech/regexp.html) and we have a couple of languages floating around in here
* It's super easy to miss a totally valid option or to let non-valid options slip through, especially when dealing with special characters and a limited subset of what's possible.